libraries/core_libs/consensus/src/pbft/sync_queue.cpp

```diff
@@ -13,7 +13,12 @@ uint64_t SyncBlockQueue::getPeriod() const {
 
 size_t SyncBlockQueue::size() const {
   std::shared_lock lock(queue_access_);
-  return queue_.size();
+  // Each sync block requires cert votes from the next block or from the last_block_cert_votes_ if it is the last block,
+  // a block is counted only if it has cert votes as well
+  if (last_block_cert_votes_.size() || queue_.size() == 0)
+    return queue_.size();
+  else
+    return queue_.size() - 1;
```

Comment:
I do not understand when this can happen? aren't we pushing block with votes all the time?